lib/ledgerBlockStorage.js

```diff
@@ -515,42 +566,32 @@ module.exports = class LedgerBlockStorage {
   _expandEvents(block, callback) {
     // find all events that must be fetched
     const events = block.event || [];
-    const eventsToFetch = {};
+    const hashes = [];
     for(let i = 0; i < events.length; ++i) {
       // TODO: Determine if we want to expand other hash types
-      const event = events[i];
-      const isEventHash = typeof(event) === 'string' &&
-        event.startsWith('ni:///');
+      const isEventHash = typeof(events[i]) === 'string' &&
+        events[i].startsWith('ni:///');
       if(isEventHash) {
-        eventsToFetch[event] = i;
+        hashes.push(events[i]);
```

Comment:
Why change using a map? The map approach avoids having to do a linear search for every single event when doing event replacement (which has now been added to the bottom).